src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantRecalibrator.java

```diff
@@ -0,0 +1,852 @@
+     *  to help describe the normalization. The model fit report can be read in with our R gsalib package. Individual
+     *  model Gaussians can be subset by the value in the "Gaussian" column if desired.
+     */
+    @Argument(fullName="output_model",
+            shortName = "outputModel",
+            doc="If specified, the variant recalibrator will output the VQSR model fit to the file specified by -modelFile or to stdout",
+            optional=true)
+    private boolean outputModel = false;
+
+    @Argument(fullName="model_file",
+            shortName = "modelFile",
+            doc="A GATKReport containing the positive and negative model fits",
+            optional=true)
+    private String modelReport = null;
+
+    @HiddenOption
+    @Argument(fullName="replicate",
+            shortName="replicate",
+            doc="Used to debug the random number generation inside the VQSR. Do not use.",
+            optional=true)
+    private int REPLICATE = 200;
+
+    /**
+     * The statistical model being built by this tool may fail due to simple statistical sampling
+     * issues. Rather than dying immediately when the initial model fails, this argument allows the
+     * tool to restart with a different random seed and try to build the model again. The first
+     * successfully built model will be kept.
+     *
+     * Note that the most common underlying cause of model building failure is that there is insufficient data to
+     * build a really robust model. This argument provides a workaround for that issue but it is
+     * preferable to provide this tool with more data (typically by including more samples or more territory)
+     * in order to generate a more robust model.
+     */
+    @Advanced
+    @Argument(fullName="max_attempts",
+            shortName = "max_attempts",
+            doc="Number of attempts to build a model before failing",
+            optional=true)
+    private int max_attempts = 1;
+
+    /////////////////////////////
+    // Debug Arguments
+    /////////////////////////////
+    @Advanced
+    @Argument(fullName = "trustAllPolymorphic",
+            shortName = "allPoly",
+            doc = "Trust that all the input training sets' unfiltered records contain only polymorphic sites to drastically speed up the computation.",
+            optional=true)
+    private Boolean TRUST_ALL_POLYMORPHIC = false;
+
+    /////////////////////////////
+    // Private Member Variables
+    /////////////////////////////
+    private VariantDataManager dataManager;
+    private VariantContextWriter recalWriter;
+    private PrintStream tranchesStream;
+    private ArrayList<Double> replicate = new ArrayList<>();
+    private final Set<String> ignoreInputFilterSet = new TreeSet<>();
+    private final VariantRecalibratorEngine engine = new VariantRecalibratorEngine( VRAC );
+    private ExpandingArrayList<VariantDatum> reduceSum = new ExpandingArrayList<>();
```

Comment:
It gets pretty large (it scales with the number of variants used by the model and the number of annotations specified). Its no different than in GATK3 though, since its just replacing the state previously kept in the map accumulator. At the end of traversal they're the same length in both GATK3 and GATK4.